continue if we encounter a module that is not provided

zingale opened the pull request:
if we file requests a module that nothing provides then continue on to the next.  This was the intended behavior originally, but we missed a `continue` statement.  This allows for system-provided
modules to work without them in the `IGNORES` list.  

This fixes a bug that @bcfriesen  ran into.

zingale committed:
if we file requests a module that nothing provides then continue
on to the next.  This was the intended behavior originally, but
we missed a `continue` statement.  This allows for system-provided
modules to work without them in the `IGNORES` list.  This fixes
a bug that Brian ran into

zingale commented:
in particular, this doesn't change the behavior we were doing, _except_ in the case that the very first file we encounter needs a module that nothing provides.  This is the case that Brian just happened to run to.  If it were the second file, the, things would have passed through without problems.

WeiqunZhang commented:
Does the white list serve any purpose now?

On Dec 1, 2016 5:43 PM, "Michael Zingale" <notifications@github.com> wrote:

> in particular, this doesn't change the behavior we were doing, *except*
> in the case that the very first file we encounter needs a module that
> nothing provides. This is the case that Brian just happened to run to. If
> it were the second file, the, things would have passed through without
> problems.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BoxLib-Codes/BoxLib/pull/18#issuecomment-264350410>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHf54TK4AF2PsPQYMqn2_r97HkohmUsZks5rD3epgaJpZM4LCFjI>
> .
>

WeiqunZhang commented:
OK I see now.  Modules in the white list will not cause and errors.  But a
missing module not in the list will generate a warning message in
f90.depends.  So I suggest the warning message be like

$(warning ....)

Then the message can be shown be the make.


Weiqun

On Thu, Dec 1, 2016 at 6:42 PM, Weiqun Zhang <weiqunzhang@lbl.gov> wrote:

> Does the white list serve any purpose now?
>
> On Dec 1, 2016 5:43 PM, "Michael Zingale" <notifications@github.com>
> wrote:
>
>> in particular, this doesn't change the behavior we were doing, *except*
>> in the case that the very first file we encounter needs a module that
>> nothing provides. This is the case that Brian just happened to run to. If
>> it were the second file, the, things would have passed through without
>> problems.
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/BoxLib-Codes/BoxLib/pull/18#issuecomment-264350410>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AHf54TK4AF2PsPQYMqn2_r97HkohmUsZks5rD3epgaJpZM4LCFjI>
>> .
>>
>

zingale committed:
use a GNU make $(warning...) to report unfound module dependencies.
We retain the print to stderr too, so we can easily see these by
simply redirecting stdout if desired

zingale commented:
good call.  I implemented that.
